### PR TITLE
[ContentUnderstanding] Add to_llm_input helper for converting analysis results to LLM-friendly text

### PR DESCRIPTION
**Pull request by chienyuanchang**
# Description

Adds the `to_llm_input()` public helper function to the `azure-ai-contentunderstanding` package. This function converts a CU `AnalysisResult` into a formatted text string (YAML front matter + markdown body) suitable for injecting into LLM prompts, storing in vector databases, or returning as tool output in agentic workflows.

**Key features:**
- Renders all content types: documents (with page markers), audio/video (with time ranges for multi-segment), and classification hierarchies (parent auto-skipped, children rendered with category labels)
- `_resolve_fields()` recursively flattens all 9 `ContentField` subtypes (`StringField`, `NumberField`, `ObjectField`, `ArrayField`, etc.) into plain Python dicts
- Span-based page markers from `pages[].spans` offsets, with `<!-- PageBreak -->` fallback for older content
- Minimal built-in YAML serializer (no external dependency) with proper quoting for dates, booleans, and YAML-special characters
- RAI warnings always included in output regardless of `include_fields`/`include_markdown` flags
- Single AV content omits `timeRange`; only multi-segment AV includes `timeRange` per segment (per design spec)
- Configurable via `include_fields`, `include_markdown`, and `metadata` keyword arguments

## Files changed

| File | Change |
|------|--------|
| `azure/ai/contentunderstanding/_helpers.py` | New file — `to_llm_input()`, `_resolve_fields()`, and supporting internal functions |
| `azure/ai/contentunderstanding/_patch.py` | Import and re-export `to_llm_input` in `__all__` |
| `tests/test_to_llm_input.py` | New file — 60 unit tests across 10 categories (public API, error handling, field resolution, documents, AV, multi-segment, classification, parameters, front matter, edge cases, real CU patterns) |

## How to verify

```bash
cd sdk/contentunderstanding/azure-ai-contentunderstanding
pip install -e .
python -m pytest tests/test_to_llm_input.py -v

### CODE REVIEW

File: sdk/contentunderstanding/azure-ai-contentunderstanding/azure/ai/contentunderstanding/_helpers.py [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+                    continue
+                tag = f"{prefix}- " if first else f"{prefix}  "
+                safe_k = _yaml_scalar(k)
+                if isinstance(v, dict) and v:
+                    lines.append(f"{tag}{safe_k}:")
+                    inner_indent = indent + 2 if first else indent + 1
+                    _emit_mapping(lines, v, inner_indent)
+                elif isinstance(v, list) and v:
+                    lines.append(f"{tag}{safe_k}:")
+                    inner_indent = indent + 1
```
[the remark this answers is not in the document]
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/azure/ai/contentunderstanding/_helpers.py]: Fixed.

File: sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py [hunk 2]
```diff
@@ -0,0 +1,889 @@
...
+    IntegerField,
+    NumberField,
+    BooleanField,
+    DateField,
+    TimeField,
+    ArrayField,
+    ObjectField,
+    JsonField,
+    ContentField,
+)
```
[the remark this answers is not in the document]
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py]: add pylint and removed unused imports

File: sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py [hunk 3]
```diff
@@ -0,0 +1,992 @@
...
+        resolved = _resolve_fields(fields)
+        assert "Addr" not in resolved
+
+    def test_array_items_with_none_filtered(self):
+        fields = {
+            "Tags": ArrayField(
+                type="array",
+                value_array=[
+                    StringField(type="string", value_string="a"),
+                    StringField(type="string", value_string=None),
```
Comment (yungshinlintw) [inline sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py]: I don't quite understand this test case. Why would we filter out None in the array?

File: sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py [hunk 4]
```diff
@@ -0,0 +1,992 @@
...
+                    StringField(type="string", value_string="b"),
+                ],
+            )
+        }
+        resolved = _resolve_fields(fields)
+        assert resolved["Tags"] == ["a", "b"]
+
+    def test_empty_fields_dict(self):
+        assert _resolve_fields({}) == {}
```
Comment (yungshinlintw) [inline sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py:258]: Please also add a test for nested object (object within object)

File: sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py [hunk 5]
```diff
@@ -0,0 +1,992 @@
...
+        doc = DocumentContent(
+            kind="document",
+            markdown="First page text.\n\nSecond page text.",
+            start_page_number=1,
+            end_page_number=2,
+            pages=[
+                DocumentPage(page_number=1, spans=[ContentSpan(offset=0, length=16)]),
+                DocumentPage(page_number=2, spans=[ContentSpan(offset=18, length=17)]),
+            ],
+        )
```
Comment (yungshinlintw) [inline sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py:288]: Please test with a document (ideally from a real multi0page pdf with simple text) that is analyzed with specific page range, such as 2, 4-5, 8-, and verify the specific markdown with page numbers.

**Comment by yungshinlintw** [comment]
Given that this is a complex feature, please also add recorded test based on test doc, including multi-page pdf with specific range, multi-segment audible/visual content, and classification result.

**Review by yungshinlintw — CHANGES_REQUESTED** [review]

File: sdk/contentunderstanding/azure-ai-contentunderstanding/azure/ai/contentunderstanding/_helpers.py
Comment (yungshinlintw) [inline sdk/contentunderstanding/azure-ai-contentunderstanding/azure/ai/contentunderstanding/_helpers.py:1]: Please also add samples for the use of to_llm_input()

**Review by yungshinlintw — CHANGES_REQUESTED** [review]

File: sdk/contentunderstanding/azure-ai-contentunderstanding/azure/ai/contentunderstanding/_helpers.py
Comment (yungshinlintw) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/azure/ai/contentunderstanding/_helpers.py:1]: Let's also add the most frequently used sample in main README.md as well

**Review by yungshinlintw — CHANGES_REQUESTED** [review]

File: sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py (hunk 3 above)
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py]: We should not have `None` in array. Removed the filter in code and this test case.

File: sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py (hunk 4 above)
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py:258]: Added `test_nested_object_within_object`

File: sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py (hunk 5 above)
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/tests/test_to_llm_input.py:288]: Added into sample.

File: sdk/contentunderstanding/azure-ai-contentunderstanding/azure/ai/contentunderstanding/_helpers.py
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/azure/ai/contentunderstanding/_helpers.py:1]: Added to_llm_input sample into `sample_analyze_binary`, `sample_analyze_invoice`, and `sample_create_classifier`, added a new sample file 'sample_to_llm_input`, and add sample code into main README. I will add test sample once we confirm the samples.

File: sdk/contentunderstanding/azure-ai-contentunderstanding/README.md [hunk 6]
```diff
@@ -517,6 +518,49 @@ async def analyze_invoice():
...
+
+```python
+from azure.ai.contentunderstanding import ContentUnderstandingClient, to_llm_input
+from azure.ai.contentunderstanding.models import AnalysisInput
+from azure.identity import DefaultAzureCredential
+
+client = ContentUnderstandingClient(endpoint, DefaultAzureCredential())
+
+poller = client.begin_analyze(
+    analyzer_id="prebuilt-invoice",
```
Comment (yungshinlintw) [inline sdk/contentunderstanding/azure-ai-contentunderstanding/README.md]: For the main README sample, let's use prebuilt-documentSearch as it's CU main solution for RAG, and let's add a document with table and diagram.

Comment (yungshinlintw) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/README.md]: It's ok to do this in the next iteration so that we can get this one in first.

File: sdk/contentunderstanding/azure-ai-contentunderstanding/samples/async_samples/sample_analyze_binary_async.py [hunk 7]
```diff
@@ -168,6 +175,19 @@ async def main() -> None:
...
+        # provides to_llm_input() which packages the result into a single text block with
+        # YAML front matter (content type, pages, fields, optional metadata) followed by
+        # the markdown body — ready for LLM prompts, vector stores, or agentic tools.
+        print("\n" + "=" * 60)
+        print("LLM-READY OUTPUT")
+        print("=" * 60)
+
+        text = to_llm_input(result)
+        print(text)
+        # [END convert_to_llm_input]
```
Comment (yungshinlintw) [inline sdk/contentunderstanding/azure-ai-contentunderstanding/samples/async_samples/sample_analyze_binary_async.py]: For the next iteration. Let's move this code block right before the output the markdown and others.

**Review by yungshinlintw — APPROVED** [review]

File: sdk/contentunderstanding/azure-ai-contentunderstanding/README.md (hunk 6 above)
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/README.md]: I can make these changes. We will still need to add the new recording sample tests.

File: sdk/contentunderstanding/azure-ai-contentunderstanding/samples/async_samples/sample_analyze_binary_async.py (hunk 7 above)
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/samples/async_samples/sample_analyze_binary_async.py]: Moved.

File: sdk/contentunderstanding/azure-ai-contentunderstanding/README.md (hunk 6 above)
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/README.md]: Updated to use local file `sample_files/sample_document_features.pdf` (maybe we could upload to assets later) with `prebuilt-documentSearch`.

File: sdk/contentunderstanding/azure-ai-contentunderstanding/tests/samples/test_sample_to_llm_input.py [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+        text = to_llm_input(result)
+        assert text.startswith("---"), "Output should start with YAML front matter"
+        assert "contentType: document" in text, "Output should declare contentType: document"
+        # The 'pages' front matter key should reflect the original page numbers (2, 3, 5),
+        # compressed by the helper as "2-3, 5" — not renumbered to 1-3 within the range.
+        assert "pages:" in text, "Output should include a 'pages' key in front matter"
+        assert "2-3, 5" in text or "'2-3, 5'" in text, (
+            f"'pages' value should be '2-3, 5' (original page numbers preserved). Output:\n{text[:500]}"
+        )
+        print(f"[PASS] to_llm_input output validated ({len(text)} chars, pages='2-3, 5' preserved)")
```
Comment (yungshinlintw) [inline sdk/contentunderstanding/azure-ai-contentunderstanding/tests/samples/test_sample_to_llm_input.py:147]: This only check the pages in the front mater.
Please also verify that the we insert the correct page nuumber in the markdown, such as: --- page 2 ---

**Review by yungshinlintw — CHANGES_REQUESTED** [review]

File: sdk/contentunderstanding/azure-ai-contentunderstanding/tests/samples/test_sample_to_llm_input.py (hunk 8 above)
Comment (chienyuanchang) [reply sdk/contentunderstanding/azure-ai-contentunderstanding/tests/samples/test_sample_to_llm_input.py:147]: Added a test.

**Review by yungshinlintw — APPROVED** [review]
